source/DSCResources/DSC_CMDistributionPoint/DSC_CMDistributionPoint.psm1

```diff
@@ -0,0 +1,612 @@
+                   ($PrimaryPackageShareLocation.Length -gt 1 -or $PrimaryPackageShareLocation -match '[0-9]') -or
+                   ($SecondaryPackageShareLocation.Lenth -gt 1 -or $SecondaryPackageShareLocation -match '[0-9]'))
+                {
+                    throw $script:localizedData.InvalidPriOrSecLetter
+                }
+
+                if (($SecondaryContentLibraryLocation -and [string]::IsNullOrEmpty($PrimaryContentLibraryLocation)) -or
+                   ($SecondaryPackageShareLocation -and [string]::IsNullOrEmpty($PrimaryPackageShareLocation)))
+                {
+                    throw $script:localizedData.SecAndNoPrimary
+                }
+
+                if ($null -eq (Get-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName))
+                {
+                    Write-Verbose -Message ($script:localizedData.SiteServerRole -f $SiteServerName)
+                    New-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName
+                }
+
+                $initialValues = @('MinimumFreeSpaceMB','PrimaryContentLibraryLocation','SecondaryContentLibraryLocation',
+                                    'PrimaryPackageShareLocation','SecondaryPackageShareLocation','CertificateExpirationTimeUtc')
+
+                foreach ($item in $initialValues)
+                {
+                    if ($PSBoundParameters.ContainsKey($item))
+                    {
+                        $dpSetupParams += @{
+                            $item = $PSBoundParameters.$item
+                        }
+                    }
+                }
+
+                if (-not $PSBoundParameters.ContainsKey('CertificateExpirationTimeUtc'))
+                {
+                    $dateValueDefault = [DateTime]::Now.AddYears(2)
+
+                    $dpSetupParams += @{
+                        CertificateExpirationTimeUtc = $dateValueDefault
+                    }
+                }
+
+                Write-Verbose -Message ($script:localizedData.AddDPRole -f $SiteServerName)
+                Add-CMDistributionPoint -SiteSystemServerName $SiteServerName -SiteCode $SiteCode  @dpSetupParams
+            }
+
+            $additionalParams = @('Description','ClientCommunicationType','AllowPreStaging','EnableAnonymous',
+                                'EnableBranchCache','EnableLedbat')
+
+            foreach ($param in $PSBoundParameters.GetEnumerator())
+            {
+                if ($additionalParams -contains $param.key)
+                {
+                    if ($param.Value -ne $state[$param.key])
+                    {
+                        Write-Verbose -Message ($script:localizedData.SettingValue -f $param.key, $param.Value)
+
+                        $buildingParams += @{
+                            $param.key = $param.Value
+                        }
+                    }
+                }
+            }
+
+            if ($BoundaryGroups)
```

Comment:
Should there be a check to ensure the boundary group exists? If I specify a boundary group that isn't there it just stays notindesiredstate.